CTP/common/ext/run_shell.sh

```diff
@@ -0,0 +1,150 @@
+coverage_controller_target_dir=""
+coverage_collaborate_url=""
+is_continue_mode=$1
+shell_config_template=""
+shell_fm_test_conf="${CTP_HOME}/conf/shell_runtime.conf"
+
+coverage_controller_ip=$MKEY_COVERAGE_UPLOAD_IP
+coverage_controller_user=$MKEY_COVERAGE_UPLOAD_USER
+coverage_controller_pwd=$MKEY_COVERAGE_UPLOAD_PWD
+coverage_controller_port=$MKEY_COVERAGE_UPLOAD_PORT
+testcase_timeout=`getMsgValue $MKEY_TESTCASE_TIMEOUT 7200`
+fail_case_retry_count=`getMsgValue $MKEY_MAX_RETRY_COUNT 0`
+db_charset=`getMsgValue $MKEY_TESTING_DEFAULT_CHARSET en_US`
+
+
+if [  "$BUILD_TYPE" == "coverage" ];then
+        role="--role-coverage"
+        coverage_controller_target_dir=${MKEY_COVERAGE_UPLOAD_DIR}/${BUILD_ID}/new
+        feedback_type="file"
+	coverage_collaborate_url=$src_url
+fi
+
+
+function run_shell()
+{
+   if [ -f ${CTP_HOME}/conf/shell_template_for_${BUILD_SCENARIOS}.conf ]; then
+      shell_config_template= ${CTP_HOME}/conf/shell_template_for_${BUILD_SCENARIOS}.conf
+   else if [ -f ${CTP_HOME}/conf/shell_template.conf ]; then
+      shell_config_template= ${CTP_HOME}/conf/shell_template.conf
+   fi
+
+   cp -f ${shell_config_template} ${shell_fm_test_conf}
+   branch=$BUILD_SCENARIO_BRANCH_GIT
+   category=$BUILD_SCENARIOS
+  
+   #init and clean log
+   tmplog=$CTP_HOME/runtime.log
+   rm $tmplog >/dev/null 2>&1 
+   
+   cd $CTP_HOME
+   #update configuration file
+   ini.sh -u "main.testcase.branch_git=$branch" $shell_fm_test_conf
+   ini.sh -u "main.testing.category=$category" $shell_fm_test_conf
+   ini.sh -u "main.testing.role=$role" $shell_fm_test_conf
+   ini.sh -u "main.collaborate.url=$coverage_collaborate_url" $shell_fm_test_conf
+   ini.sh -u "main.coverage.controller.ip=$coverage_controller_ip" $shell_fm_test_conf
+   ini.sh -u "main.coverage.controller.user=$coverage_controller_user" $shell_fm_test_conf 
+   ini.sh -u "main.coverage.controller.pwd=$coverage_controller_pwd" $shell_fm_test_conf 
+   ini.sh -u "main.coverage.controller.port=$coverage_controller_port" $shell_fm_test_conf 
+   ini.sh -u "main.coverage.controller.result=$coverage_controller_target_dir" $shell_fm_test_conf 
+   ini.sh -u "main.feedback.type=$feedback_type" $shell_fm_test_conf 
+   ini.sh -u "main.testcase.timeout=$testcase_timeout" $shell_fm_test_conf 
+   ini.sh -u "max.retry.count=$fail_case_retry_count" $shell_fm_test_conf 
+   ini.sh -u "main.testing.default_charset=$db_charset" $shell_fm_test_conf 
+   ini.sh -u "main.testbuild.url=$url" $shell_fm_test_conf
+
+   #execute testing
+   ctp.sh shell -c $shell_fm_test_conf | tee $tmplog
+}
+
+function run_shell_continue()
+{
+   #init and clean log
+   tmplog=$CTP_HOME/runtime.log
+   rm $tmplog >/dev/null 2>&1
+   
+   ini.sh -u "main.mode.continue=true" ${shell_fm_test_conf}   
+   
+   $CTP_HOME/bin/ctp.sh shell -c ${shell_fm_test_conf} | tee $tmplog
+}
+
+
+function run_shell_legacy()
+{
+    category=$BUILD_SCENARIOS
+    
+    svn_user=`ini.sh $ctp_common_conf "svn.user"`
```

Comment:
Please remove below since we didn't use them.

+    svn_user=`ini.sh $ctp_common_conf "svn.user"`
 +    svn_pwd=`ini.sh $ctp_common_conf "svn.pwd"`
 +    cubrid_common_url=`ini.sh $ctp_common_conf "cubrid.commom.url"`